### DISCUSSION TITLE
New Component Request

### DISCUSSION THREAD

**Discussion by ghost** [quasarframework/quasar#10311] [Ideas / Proposals]
Hi, can quasar team make a carousel like [slick](https://kenwheeler.github.io/slick/) with all its features and responsiveness ?

I am using quasar's [multi image slider](https://quasar.dev/vue-components/carousel#example--multi-image-slides) to make a carousel with cards inside it and its not responsive, it does not decrease the number of cards when the screen width decreases.

**Comment by ontwikkelfabriek** [comment]
@FuridamuFaita 

> I am using quasar's multi image slider to make a carousel with cards inside it and its not responsive

You could create a codpen.io with your multi card  carousel to demonstrate your issue, so we can provide support/ideas how to fix it.


Btw Please create a Feature Request in the issues section. This is the discussion forum.

**Reply by ghost** [reply]
There is already a [feature request](https://github.com/quasarframework/quasar/issues/1670) with that.

Also this is [the issue](https://github.com/quasarframework/rfcs/issues/4#issuecomment-586715437) that happens when i try to resize the screen or when viewed on phones.

**Comment by lazaroofarrill** [comment]
This is what I used in the page I'm working on right now.

```
const formattedSlides = computed(() => {
        const s = []
        let thumbs = []
        let tiles = 6
        if ($q.screen.sm) tiles = 8
        if ($q.screen.xs) tiles = 4
        for (let i = 0; i < props.slides.length; i++) {
          thumbs.push(props.slides[i])
          if (thumbs.length === (tiles)) {
            s.push({ thumbs: thumbs })
            thumbs = []
          }
        }
        if (thumbs.length) {
          s.push({ thumbs: thumbs })
        }

        return s
      })
```

```

<q-carousel
            animated
            class="text-white bg-transparent full-height full-width"
            control-color="white"
            infinite
            ref="carousel"
            swipeable
            transition-next="slide-left"
            transition-prev="slide-right"
            v-model="currentSlide">
            <q-carousel-slide :img-src="slide" :key="slide" :name="i"
                              v-for="(slide, i) in formattedSlides">
              <div class="row justify-start full-height">
                <div :key="image" class="col-6 col-sm-3 col-md-4 q-pa-sm"
                     v-for="image in slide.thumbs">
                  <q-img
                    :class="`zoom full-height ${ image !== selectedImage ? 'half-brightness' : ''}`"
                    :ratio="16/9"
                    :src="image"
                    @click="changeImage(image)"/>
                </div>
              </div>
            </q-carousel-slide>
          </q-carousel>

```


Desktop
![image](https://user-images.githubusercontent.com/35935209/128616654-959422a8-afea-400d-9db5-365601956ef9.png)


Tablet
![image](https://user-images.githubusercontent.com/35935209/128616717-d5e4ac0e-b7a4-4839-a4ed-fd54dc377ea0.png)


Mobile
![image](https://user-images.githubusercontent.com/35935209/128616661-1e9a06ee-f160-47a6-9271-8b8f81fb3e0c.png)


I believe the carousel has room for improvement, like better gesture support, but this seems like a pretty specific use case with lots of possible custom requirements.

**Reply by ghost** [reply]
That's a nice approach, I also thought about using the breakpoints.
